03-Login-app-directory/middleware.ts

```diff
@@ -0,0 +1,14 @@
+import { NextResponse } from "next/server";
+import type { NextRequest } from "next/server";
+
+export async function middleware(request: NextRequest) {
+  const authToken = request.cookies.get("psg_auth_token")?.value;
+  if (!authToken) {
+    return NextResponse.redirect(new URL("/", request.url));
+  }
```

Comment:
This only checks if `authToken` exists in the cookie. The token could be invalid and a check should be added for that.

Ideally the following should be added:
```
const authToken = request.cookies.get('psg_auth_token')?.value;

  if (!authToken) {
    return NextResponse.redirect(new URL('/auth', request.url));
  }
  const userId = await passage.validAuthToken(authToken);
  if (!userId) {
    return NextResponse.redirect(new URL('/auth', request.url));
  }
```

but for some reason, `passage.validAuthToken` is always returning `undefined` even though a valid token is present.